### PR TITLE
Clean up ready to release

### PR DESCRIPTION
**Pull request by qwandor**
I've added some tests, more doc comments, and a few other cleanups ready to publish the crate on crates.io.

(I'd like to have it published so I can depend on it from another project.)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [package]
-name = "cloudbbq2-rs"
```
Comment (ruediger) [inline Cargo.toml:2]: We could actually rename it to cloudbbq. I only named it cloudbbq2 because the first attempt was using a different bt library.

**Comment by ruediger** [comment]
Thank you for all the changes! As I suggested we could just rename this to cloudbbq. Since you are doing a lot of work, I'd be happy to make you co-owner of this repo. (We might want to still require code reviews though). WDYT?

File: Cargo.toml (hunk 1 above)
Comment (qwandor) [reply Cargo.toml:2]: Done.

**Comment by qwandor** [comment]
Sure, I'm happy to be co-owner, and agree that code review is good to have either way.

**Comment by qwandor** [comment]
I've published the crate to crates.io. If you tell me your username there I'll add you as an owner.

**Comment by ruediger** [comment]
I've logged in with my GitHub account, so I believe it's the same username: ruediger

**Pull request merged**